src/zenml/zen_server/routers/workspaces_endpoints.py

```diff
@@ -878,17 +880,19 @@ def create_run_metadata(
             "is not supported."
         )
 
-    if run_metadata.pipeline_run_id:
-        run = zen_store().get_run(run_metadata.pipeline_run_id)
+    if run_metadata.resource_type == MetadataResourceTypes.PIPELINE_RUN:
+        run = zen_store().get_run(run_metadata.resource_id)
         verify_permission_for_model(run, action=Action.UPDATE)
-
-    if run_metadata.step_run_id:
-        step_run = zen_store().get_run_step(run_metadata.step_run_id)
-        verify_permission_for_model(step_run, action=Action.UPDATE)
-
-    if run_metadata.artifact_id:
-        artifact = zen_store().get_artifact(run_metadata.artifact_id)
+    elif run_metadata.resource_type == MetadataResourceTypes.STEP_RUN:
+        step = zen_store().get_run_step(run_metadata.resource_id)
+        verify_permission_for_model(step, action=Action.UPDATE)
+    elif run_metadata == MetadataResourceTypes.ARTIFACT:
+        artifact = zen_store().get_artifact(run_metadata.resource_id)
```

Comment:
Again, I am leaving notes where the artifact version PR might affect the current implementation. You might have to get the `artifact_version` first and then check for the `artifact` within that response.

Reply:
artifact versions have their own RBAC entity so I don't think you need that